components/scream/src/physics/rrtmgp/share/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# Need to add Fortran as a language, since that is what the shr source is in
+enable_language(Fortran)
+
+# Set variables needed for processing genf90 templates
+set(CIMEROOT ${SCREAM_BASE_DIR}/../../cime)
+list(APPEND CMAKE_MODULE_PATH ${CIMEROOT}/src/CMake)
+set(GENF90 ${CIMEROOT}/src/externals/genf90/genf90.pl)
+set(ENABLE_GENF90 True)
+include(genf90_utils)
+include(Sourcelist_utils)
+
+# NOTE: I should be able to do this:
+#    # Note that this sets the share_sources variable to the list of source files we need to build
+#    add_subdirectory(${SCREAM_BASE_DIR}/../../share/util ${CMAKE_CURRENT_BINARY_DIR}/share_util_build)
+# However, that did not seem to add the fortran sources derived from the genf90 templates to the targets.
+# Instead, try explicitly adding sources and process the genf90 templates ourselves.
+# TODO: Figure out why the CMakeLists.txt and the cmake macros there do not handle this correctly.
```

Comment:
As far as I understand, CMake is not designed to dynamically generate lists of source files. Dynamically generating these lists requires a more aggressive set of criteria for regenerating CMake's makefiles. Otherwise you end up thinking you can just type `make` when in fact you must rerun `cmake` to get consistent results.

Reply:
Okay, I think I get what you’re saying. I think I was mostly upset that I could not simply do:
```
 add_subdirectory(${SCREAM_BASE_DIR}/../../share/util ${CMAKE_CURRENT_BINARY_DIR}/share_util_build)
```
but if I understand correctly that’s because of how the CMakeLists.txt is structured in that folder, and that’s probably because of the use of genf90.